server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -697,6 +705,11 @@ Number valueForSearch(Number value) {
                     (dv, n) -> new DelegateDocValuesField(new Longs(new LongsSupplier(dv)), n)
                 );
             }
+
+            @Override
+            public QueryableExpression asQueryableExpression(String field, boolean hasDocValues, SearchExecutionContext context) {
+                return INTEGER.asQueryableExpression(field, hasDocValues, context);
+            }
```

Comment:
These fields are uncommon and correctly, but inefficiently approximated as `integer` fields.